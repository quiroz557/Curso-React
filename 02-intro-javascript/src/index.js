const personajes = ['Goku','Vegeta','Trunks']

// const [ p1, p2, p3] = personajes

// console.log( p1 );
// console.log( p2 );

// si quiere solo el segundo elemento 

const [ , p2 ] = personajes
const [ , , p3] = personajes

console.log(p2, p3);

const retornaArreglo = () => {
    return ['ABC', 123]
}


const [ letras, numeros ] = retornaArreglo();

console.log(letras, numeros);

const funState = ( valor ) => {
    return [valor, () => console.log('Hola Mundo')];
}

// const state = funState( 'goku' );
// console.log( state );

// state[1]()

// TAREA
// Desestructurar el funState
// 1. El primer valor del state se llamar√° nombre
// 2. se llamara setNombre

const [ state, setState ] = funState('Goku')

console.log(state);
setState();


