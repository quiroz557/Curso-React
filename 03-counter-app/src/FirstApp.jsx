import React from 'react'
import PropTypes from 'prop-types'

// const getSaludo = ( saludo ) => `Hola ${ saludo }!!! `

export const FirstApp = ({ title, subtitle, name }) => {
    return (
        <>
            <h1 data-testid="test-title">{ title }</h1>
            <p>{ subtitle }</p>
            <p>{ subtitle }</p>
            <p>{ name }</p>
        </>
    )
}


//Estas configuraciones normalmente son al final de la inicialización y todo

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.number
}

FirstApp.defaultProps = {
    title: 'No hay título',
    subtitle: 123,
    name: 'Juan Pablo'
}