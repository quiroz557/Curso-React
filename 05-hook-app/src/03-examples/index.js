export * from './LoadingPokemon'
export * from './Pokemon'
export * from './MultipleCustomHooks'