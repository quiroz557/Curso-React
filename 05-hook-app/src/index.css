body {
    padding: 20px;
}

button {
    margin-right: 10px;
}

span {
    cursor: pointer;
}

a {
    margin-right: 5px;
}