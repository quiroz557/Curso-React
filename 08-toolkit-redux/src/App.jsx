import { useState } from 'react'
import reactLogo from './assets/react.svg'
import './App.css'
import { useDispatch, useSelector } from 'react-redux'
import { decrement, increment, incrementByAmount } from './store/slices/counter';

function App() {
  
  const { counter } = useSelector(state => state.counter);
  const dispatch = useDispatch();

  return (
    <div className="App">
      <h1>Vite + React</h1>

      <div className="card">
        <button>
          count is {  counter }
        </button>
      </div>

      <button onClick={() => dispatch( increment() )}>
          Increment
      </button>
      <button onClick={() => dispatch( decrement() )}>
        Decrement
      </button>
      <button onClick={() => dispatch( incrementByAmount(2) )}>
        IncrementBy2
      </button>
    </div>
  )
}

export default App
