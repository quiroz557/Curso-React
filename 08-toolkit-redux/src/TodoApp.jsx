import React, { useState } from 'react'
import { useGetTodoByIdQuery, useGetTodosQuery } from './store/apis/todosApi'

export const TodoApp = () => {

    const [todoId, setTodoId] = useState(1);
    const { data: todo = [], isLoading } = useGetTodoByIdQuery( todoId );
    
    const nextTodo = () => {
        setTodoId( todoId + 1 );
    }

    const prevTodo = () => {
        setTodoId( todoId - 1 );
    }

    return (
        <>
            <h1>Todos - RTK Query</h1>
            <hr />

            <h4>isLoading: {isLoading ? 'True' : 'False'}</h4>
            <pre>{ JSON.stringify(todo) }</pre>

            {/* <ul>
                {
                    todos.map(todo => (
                        <li key={todo.id}>
                            <strong>{ todo.completed ? 'Done' : 'Pending' } </strong>
                            { todo.title }
                        </li>
                    ))
                }
            </ul> */}

            <button onClick={ prevTodo }>
                Prev Todo
            </button>

            <button onClick={ nextTodo }>
                Next Todo
            </button>
        </>
    )
}
