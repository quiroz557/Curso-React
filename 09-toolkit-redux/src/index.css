body {
  padding: 50px;
  margin: 0;
  font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier new', monospace;
}