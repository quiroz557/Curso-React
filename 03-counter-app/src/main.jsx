import React from 'react';
import ReactDom from 'react-dom/client'
// import { CounterApp } from './CounterApp';
import { FirstApp } from './FirstApp';
// import { App } from './App'
import './index.css'


ReactDom.createRoot( document.getElementById('root') ).render(
    <React.StrictMode>
        <FirstApp title='titulo' />
    </React.StrictMode>
)