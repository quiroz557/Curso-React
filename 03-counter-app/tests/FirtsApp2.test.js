import { render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('Pruebas en <FirtsApp />', () => { 

  const title = 'Hola soy goku';
  const subtitle = 12345;

  test('Debe de hacer match con el snapshot', () => {

    const { container } = render(<FirstApp title={title} />)
    expect( container ).toMatchSnapshot();
  })  

  test('Debe de mostrar el mensaje "Hola, soy Goku"', () => {
    render(<FirstApp title={title} />)

    expect( screen.getByText(title) ).toBeTruthy();

  })

  test('Debe de mostrar el titulo en un H1"', () => {
    render(<FirstApp title={title} />)

    expect( screen.getByRole('heading',{ level: 1 }).innerHTML).toContain( title );

  })

  test('Debe de mostrar el subtitulo enviado por props"', () => {
    render(
        <FirstApp
            title={title}
            subtitle={ subtitle }
        />
    )

    expect( screen.getAllByText(subtitle).length).toBe(2);

  })

})