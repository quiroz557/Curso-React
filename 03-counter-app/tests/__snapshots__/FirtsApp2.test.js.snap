// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirtsApp /> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola soy goku
  </h1>
  <p>
    123
  </p>
  <p>
    123
  </p>
  <p>
    Juan Pablo
  </p>
</div>
`;
